src/libs/conduit/conduit_node.cpp

```diff
@@ -9513,6 +9513,289 @@ Node::to_double_array(Node &res) const
 
 }
 
+//---------------------------------------------------------------------------//
+/// zero copy if possible array to other array types
+//---------------------------------------------------------------------------//
+
+//---------------------------------------------------------------------------//
+void
+Node::zcip_int8_array(Node &res) const
+{
+    if(!dtype().is_int8())
+    {
```

Comment:
I think you want the compact check as well for your use case. 

I think that gets us further away from this being able to converge.  A sign we may be trying to do too much with this magic.

Reply:
I think I need to compact the data beforehand regardless in my case, but I see your point. Excluding the compact check here means that there is a small bit of magic insofar as `zcip_[type]_array` will always compact the data if a data type transform occurs, which is far from ideal.

Looking over my client code again, this "zero copy if possible" business is more endemic to the Conduit interface than I'd realized. In particular, I want to have this option in all cases where I need the data to be of a particular type/layout. Somehow, I want to tell Conduit "this is what I want the data to look like (e.g. contiguous integers); please make all the transforms necessary to get it into this layout while using as little extra memory as possible". That said, I don't think this `zcip_[type]_array` business is the answer; I'm going to go back to the drawing board on this one and see if I can come up with something more general.